src/client.ts

```diff
@@ -272,15 +220,71 @@ export class BKTClientImpl implements Bucketeer {
     this.apiClient
       .registerEvents(events, this.config.sourceId, this.config.sdkVersion)
       .catch((e) => {
-        this.saveErrorMetricsEvent(
-          this.config.featureTag,
-          e,
-          ApiId.REGISTER_EVENTS,
-        );
+        this.saveErrorMetricsEvent(this.config.featureTag, e, ApiId.REGISTER_EVENTS);
         this.config.logger?.warn('register events failed', e);
       });
   }
 
+  /**
+   * Flush all remaining events in the event store in batches.
+   * This method waits for the events to be sent before resolving.
+   * Events are sent in batches of eventsMaxQueueSize to avoid
+   * exceeding gRPC message size limits (default 2MB).
+   * @returns Promise that resolves when all events are flushed
+   */
+  private async flushAllEvents(): Promise<void> {
```

Comment:
This will send all the events in memory in batches when the destroy function is called.
Since the flush interval is now 10s, it shouldn't have too many events to be sent when shutting down.